Assets/MRTK/Providers/XRSDK/GenericXRSDKSpatialMeshObserver.cs

```diff
@@ -41,7 +41,44 @@ public class GenericXRSDKSpatialMeshObserver :
             BaseMixedRealityProfile profile = null) : base(spatialAwarenessSystem, name, priority, profile)
         { }
 
-        protected virtual bool? IsActiveLoader => true;
+        private IReadOnlyList<GenericXRSDKSpatialMeshObserver> observersCache;
```

Comment:
I am slightly worried about using a cache here. Could it handle the situation of someone enabling a new observer during runtime?

Reply:
you're right that it currently doesn't. however, in that case, I'd assume that the observers and system have already been spun up, in which case it's in the developer's hands to properly manage which ones should be enabled or not when they add a new one. these observers only really use this value to determine if they should start, but not if they should continue running.

the downside to not using a cache outweighs the benefits in my opinion, as unfortunately the method for accessing all observers allocates a list on every call.